library/std/src/primitive_docs.rs

```diff
@@ -184,9 +184,6 @@ mod prim_bool {}
 /// because `!` coerces to `Result<!, ConnectionError>` automatically.
 ///
 /// [`String::from_str`]: str::FromStr::from_str
-/// [`Result<String, !>`]: Result
-/// [`Result<T, !>`]: Result
-/// [`Result<!, E>`]: Result
```

Comment:
Are you sure that `!` isn't skipped? https://github.com/rust-lang/rust/blob/c643dd2ec8fed2852f5eee8f776d657293a6a8f2/src/librustdoc/passes/collect_intra_doc_links.rs#L1025 looks like `!` is ignored.

Reply:
Oops, you're right.

Reply:
Opened #79321.